go/enclave/db/interfaces.go

```diff
@@ -16,50 +16,38 @@ import (
 type BlockResolver interface {
 	// FetchBlock returns the L1 Block with the given hash.
 	FetchBlock(blockHash common.L1RootHash) (*types.Block, error)
+	// FetchHeadBlock - returns the head of the current chain.
+	FetchHeadBlock() (*types.Block, error)
+	// FetchLogs returns the block's logs.
+	FetchLogs(blockHash common.L1RootHash) ([]*types.Log, error)
 	// StoreBlock persists the L1 Block
 	StoreBlock(block *types.Block)
-	// ParentBlock returns the L1 Block's parent.
-	ParentBlock(block *types.Block) (*types.Block, error)
 	// IsAncestor returns true if maybeAncestor is an ancestor of the L1 Block, and false otherwise
 	IsAncestor(block *types.Block, maybeAncestor *types.Block) bool
 	// IsBlockAncestor returns true if maybeAncestor is an ancestor of the L1 Block, and false otherwise
 	// Takes into consideration that the Block to verify might be on a branch we haven't received yet
 	// Todo - this is super confusing, analyze the usage
 	IsBlockAncestor(block *types.Block, maybeAncestor common.L1RootHash) bool
-	// FetchHeadBlock - returns the head of the current chain.
-	FetchHeadBlock() (*types.Block, error)
-	// Proof - returns the block used as proof for the rollup
-	Proof(rollup *core.Rollup) (*types.Block, error)
 }
 
 type RollupResolver interface {
 	// FetchRollup returns the rollup with the given hash.
 	FetchRollup(hash common.L2RootHash) (*core.Rollup, error)
-	// FetchRollupByHeight returns the rollup with the given height.
+	// FetchRollupByHeight returns the rollup on the canonical chain with the given height.
 	FetchRollupByHeight(height uint64) (*core.Rollup, error)
-	// ParentRollup returns the rollup's parent rollup.
-	ParentRollup(rollup *core.Rollup) (*core.Rollup, error)
-	// StoreGenesisRollupHash stores the hash of the genesis rollup.
-	StoreGenesisRollupHash(rollupHash common.L2RootHash) error
+	// FetchHeadRollup returns the current head rollup of the canonical chain.
+	FetchHeadRollup() (*core.Rollup, error)
 	// StoreRollup stores a rollup.
 	StoreRollup(rollup *core.Rollup, receipts []*types.Receipt) error
-	// FetchGenesisRollup returns the rollup genesis.
-	FetchGenesisRollup() (*core.Rollup, error)
```

Comment:
This special treatment wasn't needed. We can just grab the 0th rollup using FetchRollupByHeight.